_layouts/default.html

```diff
@@ -8,7 +8,16 @@
   Free for personal and commercial use under the MIT license
   https://github.com/mmistakes/minimal-mistakes/blob/master/LICENSE
 -->
-<html lang="{{ site.locale | slice: 0,2 | default: "en" }}" class="no-js">
+{% capture locale %}
+  {% if page.lang %}
+    {{ page.lang }}
+  {% elsif site.lang %}
+    {{ site.lang }}
+  {% else %}
+    {{ site.locale | slice: 0,2 }}
+{% endcapture %}
+
+<html lang="{{ locale | default: "en" }}" class="no-js">
```

Comment:
Using `{% capture %}` is going to be problematic here: It always assigns a string to the "capture variable", while the `default` is assigned for [only `false`, `null` and empty strings][1]. Since your capture contains newlines, the default never applies.

As there's no special string conversion involved (e.g. object → string), I'd suggest that you stick to `{% assign %}` even though [it may sometimes be slower than plain `{% capture %}`s][2], which I believe doesn't matter here.

  [1]: https://shopify.github.io/liquid/filters/default/
  [2]: https://github.com/Shopify/liquid/issues/487